### PR TITLE
[#1220] Add frontend tests

### PR DESCRIPTION
**Pull request by Tejas2805**
Fixes part of #1220 

### Zoom View
* [x] Hide/Show all commit message bodies
* [x] Reloading the page should restore Zoom View

### Code View

* [x] Reloading the page should restore Code View
* [x] Hide/Show all file details
* [x] Check file types

```
Our front end testing coverage is rather low, hence UI regressions can happen 
very often.

Let's add front-end tests to prevent UI regressions.
```

### CODE REVIEW

**Comment by Tejas2805** [comment]
@jamessspanggg How did you fix the Codacy errors of Cypress not being found?

**Comment by jamessspanggg** [comment]
> @jamessspanggg How did you fix the Codacy errors of Cypress not being found?

Actually every time when add new tests, this unnecessary error occurs. Could you investigate/file for new issue based on this, i.e. whether we need to disable this check or declare our tests differently?

**Comment by Tejas2805** [comment]
@jamessspanggg there are two ways to pass the error:

1) Make the globals true.
2) "off" the rule (have shown it in comments) 

I think we can proceed with 1.

**Comment by jamessspanggg** [comment]
> I think we can proceed with 1.

Agreed.

**Comment by Tejas2805** [comment]
@jamessspanggg Something weird here. When I remove the commented rule, Codacy fails. Can check that [here](https://github.com/reposense/RepoSense/pull/1310/commits/000c7278ba16cb8b7767e5714b21406cb4d4db8f)

File: frontend/cypress/.eslintrc.json [hunk 1]
```diff
@@ -3,12 +3,11 @@
...
     "globals": {
-        "Cypress": false,
-        "cy": false,
-        "describe": false,
-        "expect": false,
-        "it": false,
-        "beforeEach": false
+        "Cypress": true,
+        "cy": true,
+        "describe": true,
```
Comment (Tejas2805) [inline frontend/cypress/.eslintrc.json:8]: I have made these so that they pass the Codacy check.

**Comment by Tejas2805** [comment]
@jamessspanggg @fzdy1914 Can I get your review on this?

File: frontend/cypress/tests/codeView/codeView_hideFileDetails.js [hunk 2]
```diff
@@ -0,0 +1,103 @@
+describe('hide all file details', () => {
```
Comment (jamessspanggg) [inline frontend/cypress/tests/codeView/codeView_hideFileDetails.js:1]: not sure if we need tests on hiding file details as it is not a crucial feature, the test might be taking up time instead

File: frontend/cypress/tests/zoomView/zoomView_hideCommitMessages.js [hunk 3]
```diff
@@ -0,0 +1,60 @@
+describe('hide all commit messages ', () => {
```
Comment (jamessspanggg) [inline frontend/cypress/tests/zoomView/zoomView_hideCommitMessages.js:1]: same for this

**Review by jamessspanggg — COMMENTED** [review]
Some comments:

File: frontend/cypress/tests/codeView/codeView_hideFileDetails.js (hunk 2 above)
Comment (jamessspanggg) [reply frontend/cypress/tests/codeView/codeView_hideFileDetails.js:1]: @fzdy1914 ur opinion on this?

Comment (fzdy1914) [reply frontend/cypress/tests/codeView/codeView_hideFileDetails.js:1]: I am OK with this testing actually. How much time will this extra test need?

Comment (Tejas2805) [reply frontend/cypress/tests/codeView/codeView_hideFileDetails.js:1]: Together both the  "hide" tests take a total of approximately 5000 ms, so 5 seconds

Comment (fzdy1914) [reply frontend/cypress/tests/codeView/codeView_hideFileDetails.js:1]: The time needed is OK for me.

**Review by jamessspanggg — APPROVED** [review]

**Pull request merged**